### PR TITLE
Loging implementation

### PR DESCRIPTION
**Pull request by ishaque-mobikasa**
Login screen with validation and Obscuring password

### CODE REVIEW

File: lib/core/routes.dart [hunk 1]
```diff
@@ -26,10 +28,10 @@ class AppPages {
     //   name: Routes.homeScreen,
     //   page: () => const HomeScreen(),
     // ),
-    // GetPage(
-    //   name: Routes.loginScreen,
```
Comment (tusharmobikasa) [inline lib/core/routes.dart:30]: Please reformat the code and removed unused code

File: lib/app/app_colors.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter/material.dart';
```
Comment (tusharmobikasa) [inline lib/app/app_colors.dart:1]: Wrap with a class

File: lib/app/decorations.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import 'package:flutter/material.dart';
+import 'package:task/app/dimensions.dart';
```
Comment (tusharmobikasa) [inline lib/app/decorations.dart:2]: why not using a class ??

File: lib/main.dart [hunk 4]
```diff
@@ -14,12 +15,10 @@ class MyApp extends StatelessWidget {
       debugShowCheckedModeBanner: false,
       title: 'Tasks',
       getPages: AppPages.pages,
-
```
Comment (tusharmobikasa) [inline lib/main.dart:17]: Why initial route is login where is splash screen route ? and please do not make regular changes in main.dart

File: lib/presentation/pages/login/login_view.dart [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            padding: const EdgeInsets.only(top: 8.0),
+            child: Column(children: [
+              Container(
+                decoration: const BoxDecoration(
+                    image: DecorationImage(
+                        image: AssetImage("assets/images/login.jpg"),
+                        fit: BoxFit.cover)),
+                height: size.height * 0.4,
+              ),
+              const SizedBox(
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/login_view.dart:36]: Make a class for size box widget to refactor the code and pass value of height and width with optional parameters

File: lib/presentation/pages/login/login_view.dart [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                height: 15,
+              ),
+              const Padding(
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/login_view.dart:39]: if using dimens class take the value from that class only

File: lib/presentation/pages/login/widgets/custom_form_field.dart [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+              border: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(4)))),
+          validator: validator ??
+              (type == FieldType.eMail
+                  ? (value) {
+                      if (value!.isEmpty) {
+                        return "$hintText Cant be empty";
+                      } else if (value.length < 4) {
+                        return "$hintText Must be greater than 4 charactors long";
+                      } else if (!value.isValidEmail &&
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/widgets/custom_form_field.dart:96]: Why used regex to make condition check and already made extension for phone number

File: lib/presentation/pages/login/widgets/custom_form_field.dart [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+                              .hasMatch(value)) {
+                        return "Invalid Email/Phone Number";
+                      } else {
+                        return null;
+                      }
+                    }
+                  : type == FieldType.password
+                      ? (value) {
+                          if (value == '' || value == null) {
+                            return "$hintText Cant be empty";
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/widgets/custom_form_field.dart:107]: Why 2 else if used can be made in one also

File: pubspec.lock [hunk 9]
```diff
@@ -36,6 +36,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.16.0"
```
Comment (tusharmobikasa) [inline pubspec.lock:38]: why this file is in the commit?

**Review by tusharmobikasa — COMMENTED** [review]
Please reformate the code and don't push unwanted files also try to refactor the code and make less use of else if,use  class and then declare the variable ,and why on boarding code is there if its already merged

File: lib/core/routes.dart (hunk 1 above)
Comment (ishaque-mobikasa) [reply lib/core/routes.dart:30]: Removed the commented lines.. And formatted with VS code default formatting

File: lib/app/app_colors.dart (hunk 2 above)
Comment (ishaque-mobikasa) [reply lib/app/app_colors.dart:1]: wrapped with CustomColor class

File: lib/app/decorations.dart (hunk 3 above)
Comment (ishaque-mobikasa) [reply lib/app/decorations.dart:2]: Used the dimensions class

File: lib/main.dart (hunk 4 above)
Comment (ishaque-mobikasa) [reply lib/main.dart:17]: changed to splash as initial route

File: lib/presentation/pages/login/login_view.dart (hunk 5 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/login_view.dart:36]: ok

File: lib/presentation/pages/login/login_view.dart (hunk 6 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/login_view.dart:39]: Used the dimensions class values

File: lib/presentation/pages/login/widgets/custom_form_field.dart (hunk 7 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/widgets/custom_form_field.dart:96]: Replaced with the regX

File: lib/presentation/pages/login/widgets/custom_form_field.dart (hunk 8 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/widgets/custom_form_field.dart:107]: changed 1 unwanted elseif check

File: pubspec.lock (hunk 9 above)
Comment (ishaque-mobikasa) [reply pubspec.lock:38]: added puspec.lock to gitignore

**Pull request merged**